src/aspire/basis/fle_2d.py

```diff
@@ -296,12 +306,13 @@ def _compute_nufft_points(self):
             * np.arange(self.num_angular_nodes // 2, dtype=self.dtype)
             / self.num_angular_nodes
         )
-        x = np.cos(phi).reshape(1, self.num_angular_nodes // 2)
-        y = np.sin(phi).reshape(1, self.num_angular_nodes // 2)
-        x = x * nodes * h
-        y = y * nodes * h
-        self.grid_x = x.flatten()
-        self.grid_y = y.flatten()
+        grid_xy = np.empty(
+            (2, self.num_radial_nodes, self.num_angular_nodes // 2), dtype=self.dtype
+        )
+        grid_xy[0] = np.cos(phi)  # x
+        grid_xy[1] = np.sin(phi)  # y
+        grid_xy *= nodes * h
```

Comment:
Might as well make the grid precursors on GPU too (and replace this in-place op).